x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/SecurityRestFilter.java

```diff
@@ -102,13 +102,15 @@ public void handleRequest(RestRequest request, RestChannel channel, NodeClient c
             SSLEngineUtils.extractClientCertificates(logger, threadContext, httpChannel);
         }
 
-        authenticationService.authenticate(maybeWrapRestRequest(request), ActionListener.wrap(authentication -> {
+        final RestRequest wrappedRequest = maybeWrapRestRequest(request);
+        authenticationService.authenticate(wrappedRequest, ActionListener.wrap(authentication -> {
             if (authentication == null) {
                 logger.trace("No authentication available for REST request [{}]", request.uri());
             } else {
                 logger.trace("Authenticated REST request [{}] as {}", request.uri(), authentication);
             }
-            secondaryAuthenticator.authenticateAndAttachToContext(request, ActionListener.wrap(secondaryAuthentication -> {
+            authenticationService.auditSuccess(wrappedRequest);
```

Comment:
This is where successful HTTP authentication is audtited (even though it actually happens sooner).

Reply:
The code might be easier to follow if we make `AuditTrailService` available in `SecurityRestFilter` and directly audit success here instead of delegating to authenticationService and also secondaryAuthenticator. For comparison, the securityActionFilter already has a reference for auditTrail.

I don't have a lot context of this project. So appologies if this has already been discussed and disregarded. Thanks!

Reply:
Ok, pushed [39717a7](https://github.com/elastic/elasticsearch/pull/94120/commits/39717a76001c3d7f3556a1136d804cecfede8318)